File: cheaters/PacketLife/nitro-python-1.0/apidoc/docs/samples/ns-csharp-sample/csw.cs
using System;
using System.Runtime.Remoting;
using System.Xml.Serialization;
using System.Web.Services;
    
	namespace NSConfig {
		public class ClientService : NSConfigService

                {
                        private string cookie = null;
                        /* override the getWebRequest to send cookie */
                        protected override System.Net.WebRequest  GetWebRequest(Uri uri)
                        {
                                System.Net.HttpWebRequest req = (System.Net.HttpWebRequest) base.GetWebRequest(uri);
                                if (cookie != null)
                                {
                                        req.Headers.Add("Set-Cookie", cookie);
                                }
                                return req;
                        }
                        /* override the getWebResponse to get the cookie */

                        protected override System.Net.WebResponse GetWebResponse(System.Net.WebRequest req)
                        {
                               System.Net.HttpWebResponse rep = (System.Net.HttpWebResponse) base.GetWebResponse(req);
                               if (rep.Headers["Set-Cookie"] != null)
                                {
                                       cookie = rep.Headers["Set-Cookie"];
                                }
                                return rep;
			}
                        public ClientService()
                        {
                                // Change the IP address pointing to netscaler.
                                this.Url = "http://10.102.4.111:18000";
                        }
                        public ClientService(string servername)
                        {
                                this.Url = "http://"+servername+"/soap";;
                        }
                }
	
	class csw 
	{
 		static ClientService client=null;
 
		[STAThread]
		static void Main(string[] args)
		{
			if ( args.Length < 3)
			{
				Console.WriteLine("Usage: getConfig <NS IP> username password");
				return;
			}
			string serverip = args[0];
			string username = args[1];
			string password = args[2];
			try {
          			Console.WriteLine("\nConnecting to server "+serverip+" ............\n");
				client = new ClientService(serverip);

				client.CookieContainer = new System.Net.CookieContainer();
				simpleResult result = client.login(username,password) ;
				Console.WriteLine("login : "+result.message);
       
				result = client.enablensfeature(featureEnum.LB);
				Console.WriteLine("enablensfeature	LB : " + result.message);

				result = client.enablensfeature(featureEnum.CS);
				Console.WriteLine("enablensfeature	CS : " + result.message);

				result = client.addserver("s1","10.102.3.91",null,enabledisabledEnum.VALNOTSET);
				Console.WriteLine("addserver	s1 : " + result.message);

				result = client.addserver("s2","10.102.3.92",null,enabledisabledEnum.VALNOTSET);
				Console.WriteLine("addserver	s2 : " + result.message);

				result = client.addserver("s3","10.102.3.93",null,enabledisabledEnum.VALNOTSET);
				Console.WriteLine("addserver	s3 : " + result.message);

				result = client.addservice("srv1",null,"s1",servicetypeEnum.HTTP,80,0xFFFFFFFF,cachtypeEnum.VALNOTSET,enabledisabledEnum.VALNOTSET);
				Console.WriteLine("addservice	srv1 : " + result.message);

				result = client.addservice("srv2",null,"s2",servicetypeEnum.HTTP,80,0xFFFFFFFF,cachtypeEnum.VALNOTSET,enabledisabledEnum.VALNOTSET);
				Console.WriteLine("addservice	srv2 : " + result.message);

				result = client.addservice("srv3",null,"s3",servicetypeEnum.HTTP,80,0xFFFFFFFF,cachtypeEnum.VALNOTSET,enabledisabledEnum.VALNOTSET);
				Console.WriteLine("addservice	srv3 : " + result.message);

				result = client.addlbvserver("lbvip1",vservicetypeEnum.HTTP,null,0xFFFFFFFF,0xFFFFFFFF,enabledisabledEnum.VALNOTSET,null);
				Console.WriteLine("addlbvserver	lbvip1 : " + result.message);

				result = client.addlbvserver("lbvip2",vservicetypeEnum.HTTP,null,0xFFFFFFFF,0xFFFFFFFF,enabledisabledEnum.VALNOTSET,null);
				Console.WriteLine("addlbvserver	lbvip2 : " + result.message);

				result = client.addlbvserver("lbvip3",vservicetypeEnum.HTTP,null,0xFFFFFFFF,0xFFFFFFFF,enabledisabledEnum.VALNOTSET,null);
				Console.WriteLine("addlbvserver	lbvip3 : " + result.message);

				result = client.bindlbvserver_service("lbvip1","srv1",0xFFFFFFFF);
				Console.WriteLine("bindlbvserver_service	lbvip1 : " + result.message);

				result = client.bindlbvserver_service("lbvip2","srv2",0xFFFFFFFF);
				Console.WriteLine("bindlbvserver_service	lbvip2 : " + result.message);

				result = client.bindlbvserver_service("lbvip3","srv3",0xFFFFFFFF);
				Console.WriteLine("bindlbvserver_service	lbvip3 : " + result.message);

				result = client.addcspolicy("contains_html",null,"REQ.HTTP.URL CONTAINS html",null);
				Console.WriteLine("addcspolicy	contains_html : " + result.message);

				result = client.addcspolicy("contains_txt",null,"REQ.HTTP.URL CONTAINS gif",null);
				Console.WriteLine("addcspolicy	contains_txt : " + result.message);

				result = client.addcspolicy("contains_gif",null,"REQ.HTTP.URL CONTAINS asp",null);
				Console.WriteLine("addcspolicy	contains_gif : " + result.message);

				result = client.addcsvserver("csvip",csvservicetypeEnum.HTTP,"10.102.3.54",0xFFFFFFFF,80,enabledisabledEnum.VALNOTSET,null);
				Console.WriteLine("addcsvserver	csvip : " + result.message);

				result = client.bindcsvserver_targetvserver("csvip","lbvip1","contains_html",5,null,vserverbindpointEnum.VALNOTSET,false,policylabelinvoketypeEnum.VALNOTSET,null);
				Console.WriteLine("bindcsvserver_targetvserver	csvip : " + result.message);

				result = client.bindcsvserver_targetvserver("csvip","lbvip2","contains_txt",5,null,vserverbindpointEnum.VALNOTSET,false,policylabelinvoketypeEnum.VALNOTSET,null);
				Console.WriteLine("bindcsvserver_targetvserver	csvip : " + result.message);

				result = client.bindcsvserver_targetvserver("csvip","lbvip3","contains_gif",5,null,vserverbindpointEnum.VALNOTSET,false,policylabelinvoketypeEnum.VALNOTSET,null);
				Console.WriteLine("bindcsvserver_targetvserver	csvip : " + result.message);

				result = client.logout();
				Console.WriteLine("logout : " + result.message); 
			}
			catch (Exception ex) {
			      Console.WriteLine( "Exception: "+ex.ToString());
			}
		}

	}
	}

File: cheaters/PacketLife/nitro-python-1.0/apidoc/docs/samples/ns-java-sample/csw.java
package NSConfig;
import java.io.*;
import java.util.*;
import java.net.*;
import java.beans.*;
import java.net.*;
import NSConfig.*;
import org.apache.axis.types.*;

public class csw  {
	public static void main(String[] args) {
	   String  soap_server;
	   String username;
	   String password ;

	   NSConfigBindingStub client ;
	   SimpleResult  result;
	   UnsignedInt NULL_UINT=null;
	   if (args.length < 3) {
	      System.out.println("Usage : java setConfig.class <NP> <username> <password>");
	      return;
	   }
	   soap_server = args[0];
	   username    = args[1];
	   password    = args[2];
	   try {
            			// Initialized the client stub
				soap_server = "http://"+soap_server+"/soap";
				System.out.println("Connecting to  "+soap_server + " ........");

				client  = new NSConfigBindingStub(new URL(soap_server),null);
			    client.setMaintainSession(true);
				((org.apache.axis.client.Stub)client)._setProperty(org.apache.axis.AxisEngine.PROP_DOMULTIREFS, Boolean.FALSE);
				//Login
				result = client.login(username, password);
				System.out.println("Login : "+result.getMessage());

				result = client.enablensfeature((FeatureEnum)FeatureEnum.LB);
				System.out.println("Enablensfeature	LB : " + result.getMessage());

				result = client.enablensfeature((FeatureEnum)FeatureEnum.CS);
				System.out.println("Enablensfeature	CS : " + result.getMessage());

				result = client.addserver("s1","10.102.3.91",null,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET);
				System.out.println("Addserver	s1 : " + result.getMessage());

				result = client.addserver("s2","10.102.3.92",null,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET);
				System.out.println("Addserver	s2 : " + result.getMessage());

				result = client.addserver("s3","10.102.3.93",null,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET);
				System.out.println("Addserver	s3 : " + result.getMessage());

				result = client.addservice("srv1",null,"s1",(ServicetypeEnum)ServicetypeEnum.HTTP,new UnsignedInt(80),null,(CachtypeEnum)CachtypeEnum.VALNOTSET,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET);
				System.out.println("Addservice	srv1 : " + result.getMessage());

				result = client.addservice("srv2",null,"s2",(ServicetypeEnum)ServicetypeEnum.HTTP,new UnsignedInt(80),null,(CachtypeEnum)CachtypeEnum.VALNOTSET,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET);
				System.out.println("Addservice	srv2 : " + result.getMessage());

				result = client.addservice("srv3",null,"s3",(ServicetypeEnum)ServicetypeEnum.HTTP,new UnsignedInt(80),null,(CachtypeEnum)CachtypeEnum.VALNOTSET,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET);
				System.out.println("Addservice	srv3 : " + result.getMessage());

				result = client.addlbvserver("lbvip1",(VservicetypeEnum)VservicetypeEnum.HTTP,null,null,null,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET,null);
				System.out.println("Addlbvserver	lbvip1 : " + result.getMessage());

				result = client.addlbvserver("lbvip2",(VservicetypeEnum)VservicetypeEnum.HTTP,null,null,null,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET,null);
				System.out.println("Addlbvserver	lbvip2 : " + result.getMessage());

				result = client.addlbvserver("lbvip3",(VservicetypeEnum)VservicetypeEnum.HTTP,null,null,null,(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET,null);
				System.out.println("Addlbvserver	lbvip3 : " + result.getMessage());

				result = client.bindlbvserver_service("lbvip1","srv1",null);
				System.out.println("Bindlbvserver_service	lbvip1 : " + result.getMessage());

				result = client.bindlbvserver_service("lbvip2","srv2",null);
				System.out.println("Bindlbvserver_service	lbvip2 : " + result.getMessage());

				result = client.bindlbvserver_service("lbvip3","srv3",null);
				System.out.println("Bindlbvserver_service	lbvip3 : " + result.getMessage());

				result = client.addcspolicy("contains_html",null,"REQ.HTTP.URL CONTAINS html",null);
				System.out.println("Addcspolicy	contains_html : " + result.getMessage());

				result = client.addcspolicy("contains_txt",null,"REQ.HTTP.URL CONTAINS gif",null);
				System.out.println("Addcspolicy	contains_txt : " + result.getMessage());

				result = client.addcspolicy("contains_gif",null,"REQ.HTTP.URL CONTAINS asp",null);
				System.out.println("Addcspolicy	contains_gif : " + result.getMessage());

				result = client.addcsvserver("csvip",(CsvservicetypeEnum)CsvservicetypeEnum.HTTP,"10.102.3.54",null,new UnsignedInt(80),(EnabledisabledEnum)EnabledisabledEnum.VALNOTSET,null);
				System.out.println("Addcsvserver	csvip : " + result.getMessage());

				result = client.bindcsvserver_targetvserver("csvip","lbvip1","contains_html",new UnsignedInt(5),null,(VserverbindpointEnum)VserverbindpointEnum.VALNOTSET,false,(PolicylabelinvoketypeEnum)PolicylabelinvoketypeEnum.VALNOTSET,null);
				System.out.println("Bindcsvserver_targetvserver	csvip : " + result.getMessage());

				result = client.bindcsvserver_targetvserver("csvip","lbvip2","contains_txt",new UnsignedInt(5),null,(VserverbindpointEnum)VserverbindpointEnum.VALNOTSET,false,(PolicylabelinvoketypeEnum)PolicylabelinvoketypeEnum.VALNOTSET,null);
				System.out.println("Bindcsvserver_targetvserver	csvip : " + result.getMessage());

				result = client.bindcsvserver_targetvserver("csvip","lbvip3","contains_gif",new UnsignedInt(5),null,(VserverbindpointEnum)VserverbindpointEnum.VALNOTSET,false,(PolicylabelinvoketypeEnum)PolicylabelinvoketypeEnum.VALNOTSET,null);
				System.out.println("Bindcsvserver_targetvserver	csvip : " + result.getMessage());

				result = client.logout();
				System.out.println("logout : " + result.getMessage());
	   }
	   catch (Exception ex) {
	       ex.printStackTrace();
	   }
	}

}
